tfe-scripts/tf-add-app-run.sh

```diff
@@ -37,23 +41,39 @@ napespace_apps_data_to_update="$(jq -n --arg new_value "$napespace_apps_new_valu
 
 namespace_apps_id="$(update_var "$TFC_WORKSPACE_ID" "$namespace_apps_id" "$napespace_apps_data_to_update" | jq -r '.data.id')"
 
-echo "$namespace_apps_id"
-
 # jq will ensure that the value is properly quoted and escaped to produce a valid JSON string.
 # shellcheck disable=SC2016
 run_payload="$(jq -n --arg workspace_id "$TFC_WORKSPACE_ID" '{"data":{"type":"runs","relationships":{"workspace":{"data":{"type":"workspaces","id":$workspace_id}}}}}')"
 
 run_id="$(create_run "$run_payload" | jq -r '.data.id')"
 
+if [ "$run_id" == null ]; then
+  echo "failed to create a run"
+  exit 1
+fi
+
+echo "run $run_id created"
+
+# see https://github.com/hashicorp/go-tfe/blob/master/run.go#L49 for all available run statuses.
+completed_statuses=(null "applied" "canceled" "discarded" "errored" "planned_and_finished" "policy_soft_failed")
+
+count=0
 get_status() {
   status="$(get_run "$run_id" | jq -r '.data.attributes.status')"
+  echo "run status - $status"
+
+  # disable shellcheck to match literally
+  # shellcheck disable=SC2076
+  if [[ "${completed_statuses[@]}" =~ "$status" ]]; then
+    return 1
+  fi
```

Comment:
should this script `exit 1` if the completed satus is `errored`?

Reply:
It is a very good question. my initial intent was not to interfere CI/CD pipelines in case that the ocp token is expired.
Since we have a permanent token set on TFC workspace, it should be okay to exit 1 now then.